File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.py
def solution(data, col, row_begin, row_end):
    # 1. 정렬
    data.sort(key=lambda x: (x[col-1], -x[0]))

    # 2. S_i 계산 및 XOR 누적
    answer = 0
    for i in range(row_begin, row_end + 1):
        row = data[i - 1]  # i번째 행 (인덱스는 i-1)
        S_i = sum(value % i for value in row)
        answer ^= S_i  # XOR 연산

    return answer

File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.js
function solution(data, col, row_begin, row_end) {
  // 1. 정렬
  data.sort((a, b) => {
    if (a[col - 1] === b[col - 1]) return b[0] - a[0];
    return a[col - 1] - b[col - 1];
  });

  // 2. S_i 계산 및 XOR 누적
  let answer = 0;
  for (let i = row_begin; i <= row_end; i++) {
    const row = data[i - 1];
    const S_i = row.reduce((sum, value) => sum + (value % i), 0);
    answer ^= S_i;
  }

  return answer;
}
